gnssanalysis/gn_io/igslog.py

```diff
@@ -109,21 +134,25 @@ def parse_igs_log(filename_array: _np.ndarray) -> _np.ndarray:
     with open(file_path, "rb") as file:
         data = file.read()
 
-    blk_id = _REGEX_ID.search(data)
+    blk_id = _REGEX_ID_V2.search(data)
     if blk_id is None:
-        tqdm.write(f"ID rejected from {file_path}")
-        return _np.array([]).reshape(0, 12)
+        blk_id = _REGEX_ID.search(data)
+        if blk_id is None:
+            tqdm.write(f"ID rejected from {file_path}")
+            return _np.array([]).reshape(0, 12)
 
     blk_id = [blk_id[1].decode().upper(), blk_id[2].decode().upper()]  # no .groups() thus 1 and 2
     code = blk_id[0]
     if code != file_code:
         tqdm.write(f"{code}!={file_code} at {file_path}")
         return _np.array([]).reshape(0, 12)
 
-    blk_loc = _REGEX_LOC.search(data)
+    blk_loc = _REGEX_LOC_V2.search(data)
     if blk_loc is None:
```

Comment:
if we have a variable version2 we can just do 
```
if version2:
  do something
else:
  do something else

Reply:
Achieved with new function search for version number, and then assigning existing variables based on the regex pattern to use